platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkMetrics.java

```diff
@@ -177,17 +196,22 @@ public void update() {
      *
      * @return the average times, for each member, in milliseconds
      */
-    public double[] getPingMilliseconds() {
-        final double[] times = new double[avgPingMilliseconds.size()];
-        for (int i = 0; i < times.length; i++) {
-            times[i] = avgPingMilliseconds.get(i).get();
-        }
-        times[selfId.getIdAsInt()] = 0;
+    @NonNull
+    public Map<NodeId, Double> getPingMilliseconds() {
+        final Map<NodeId, Double> times = new HashMap<>();
+        avgPingMilliseconds.forEach((nodeId, metric) -> times.put(nodeId, metric.get()));
+        times.put(selfId, 0.0);
         return times;
     }
 
-    public List<RunningAverageMetric> getAvgPingMilliseconds() {
-        return avgPingMilliseconds;
+    /**
+     * Returns the average number of bytes sent per second to each member.
+     *
+     * @return the average number of bytes sent per second to each member
+     */
+    @NonNull
+    public Map<NodeId, RunningAverageMetric> getAvgPingMilliseconds() {
+        return new HashMap<>(avgPingMilliseconds);
```

Comment:
Would it make sense to return an unmodifiable view of this map, rather than making a copy each time?

Reply:
You are right.   I think I was thinking about capturing the contents of the map in an unchanging state.  But the object references to the metrics don't change.  It's their internal state that is being updated.   Returning the map of metrics breaks the encapsulation somewhat.   Someone could spuriously update the metric.   The code path getting the map is marked for deletion, but I am going to change the method being called to return Map<NodeId, Double> to not leak the metrics references.